app/main.py

```diff
@@ -2,24 +2,69 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> None:
```

Comment:
the `__get__`,` __set__`, and other methods have general object data types. This reduces the readability and efficiency of type annotations. Define the data types more clearly, for example:
`instance: object, owner: type`